lib/dpl/provider/bintray.rb

```diff
@@ -0,0 +1,433 @@
+require 'json'
+require 'net/http'
+require "uri"
+require 'find'
+
+module DPL
+  class Provider
+    class Bintray < Provider
+
+      @file
+      @user
+      @key
+      @passphrase
+      @url
+      @descriptor
+
+      def check_auth
+      end
+
+      def needs_key?
+        false
+      end
+
+      def initFromArgs
+        @user = options[:user]
+        @key = options[:key]
+        @url = options[:url]
+        @file = options[:file]
+        @passphrase = options[:passphrase]
+
+        if @user.nil?
+          abort("The 'user' argument is required")
+        end
+        if @key.nil?
+          abort("The 'key' argument is required")
+        end
+        if @file.nil?
+          abort("The 'file' argument is required")
+        end
+        if @url.nil?
+          @url = 'https://api.bintray.com'
+        end
+      end
+
+      def readDescriptor
+        log "Reading descriptor file: #{@file}"
+        file = File.open(@file)
+        content = ""
+        file.each {|line|
+          content << line
+        }
+
+        @descriptor = JSON.parse(content)
```

Comment:
Seems a rather roundabout way to read the content of `@file`. You can just do `JSON.parse(File.read(@file))`.